hack/prow/run_tests.py

```diff
@@ -0,0 +1,140 @@
+
+"""This script will execute a set of named minikube tests,
+   gather the results, logs, and artifacts into a named GCS
+   bucket for presentation in k8s testgrid:
+   https://k8s-testgrid.appspot.com
+"""
+
+import os, sys, json, re, argparse, calendar, time, subprocess, shlex
+
+def write_results(outdir, started, finished, test_results):
+  """ write current results into artifacts/junit_runner.xml 
+      format:
+      <testsuite failures="XXX" tests="YYY" time="ZZZ">
+         <testcase classname="SUITENAME" name="TESTNAME" time="ZZZ1" />
+         <testcase classname="SUITENAME" name="TESTNAME" time="ZZZ1" />
+         ...
+      </testsuite>
+
+      write the started.json and finish.json files 
+      format:
+      started.json: {"timestamp":STARTTIMEINSECONDSINCEEPOCH}
+      finished.json: {"timestamp":FINISHTIMEINSECONDSINCEEPOCH,
+                      "passed":FINALRESULT,
+                      "result":SUCCESS|FAIL,
+                      "metadata":{}
+                      }
+  """
+  started_json_file = open('%s/started.json' % outdir, 'w')
+  finished_json_file = open('%s/finished.json' % outdir, 'w')
+  junit_xml_file = open('%s/artifacts/junit_runner.xml' % outdir, 'w')
+
+  failures = 0
+  testxml = ""
+  for test in test_results:
+    testxml += '<testcase classname="%s" name="%s" time="%s">' % (test['classname'], test['name'], test['time'])
+    if test['status'] == 'FAIL':
+        failures += 1
+        testxml += '<failure message="Test Failed" />'
+    testxml += '</testcase>\n'
+  junit_xml_file.write('<testsuite failures="%s" tests="%s">\n' % (failures, len(test_results)))
+  junit_xml_file.write(testxml)
+  junit_xml_file.write('</testsuite>')
+  junit_xml_file.close()
+
+  started_json_file.write(json.dumps(started)) 
+  started_json_file.close()
+  finished_json_file.write(json.dumps(finished)) 
+  finished_json_file.close()
+
+  return
+
+def upload_results(outdir, test, buildnum, bucket):
+  """ push the contents of gcs_out/* into bucket/test/logs/buildnum"""
+  classname = os.path.basename(test).split('.')[0]
+  args = shlex.split("gsutil cp -R gcs_out/ gs://%s/logs/%s/%s" % (bucket, classname, buildnum))
+  p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+  for line in p.stdout:
+    print line
+
+def gather_artifacts():
+  """ gather any named or default artifacts into artifacts/ """
+  return
+
+def run_tests(test, build_log, exit_status, started, finished, test_results):
+  """ execute the test, grab the start time, finish time, build logs and exit status
+      Pull test results and important information out of the build log 
+      test results format should be:
+      === RUN   TestFunctional/Mounting
+      --- PASS: TestFunctional (42.87s)
+          --- PASS: TestFunctional/Status (2.33s)
+          --- FAIL: SOMETESTSUITE/TESTNAME (seconds)
+  """
+  classname = os.path.basename(test).split('.')[0]
+  build_log_file = open(build_log, 'w')
```

Comment:
If you were to ask me the difference between build_log and build_log_file, I wouldn't be able to articulate it. Could you find more meaningful names? Perhaps:

log_path, log_file or lpath, lf